wayland-display-core/src/wayland/handlers/relative_pointer.rs

```diff
@@ -1,5 +1,17 @@
 use smithay::delegate_relative_pointer;
+use smithay::delegate_pointer_constraints;
+use smithay::input::pointer::PointerHandle;
+use smithay::wayland::output::OutputHandler;
+use smithay::wayland::pointer_constraints::PointerConstraintsHandler;
 
 use crate::comp::State;
 
-delegate_relative_pointer!(State);
\ No newline at end of file
+impl OutputHandler for State {}
+impl PointerConstraintsHandler for State{
+    fn new_constraint(&mut self, surface: &WlSurface, pointer: &PointerHandle<Self>) {
+        todo!()
```

Comment:
I guess this is what needs to be implemented in order to support the pointer constraints protocol; not sure what is expected in that method..

Reply:
Store that in the state somewhere and then obey the constraint in the input code.

See the PR that implemented this in anvil: https://github.com/Smithay/smithay/pull/872/commits/d978a2b

Also we need to constuct the `PointerConstraintsState` in `src/comp/mod.rs:init` or we never advertise support for it.

(And it would be nice, if this would be put into a separate file like all the other protocol handlers.)